packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -231,10 +230,9 @@ export class TokenDetectionController extends StaticIntervalPollingController<
     this.#disabled = disabled;
     this.setIntervalLength(interval);
 
-    this.#selectedAddress =
-      selectedAddress ??
-      this.messagingSystem.call('AccountsController:getSelectedAccount')
-        .address;
+    this.#selectedAccountId = this.messagingSystem.call(
```

Comment:
What about adding a `#getSelectedAccount` like for other controllers in this PR:

```ts
  #getSelectedAccount() {
    return this.messagingSystem.call('AccountsController:getSelectedAccount');
  }
```